### DISCUSSION TITLE
u-p formulation for HM coupled analyses under dynamic conditions

### DISCUSSION THREAD

**Discussion by nlosacco** [idaholab/moose#23446] [Q&A Modules: General] [answered]
Hi all,

I have looked through MOOSE docs but I couldn't understand if it is possible to run coupled hydro-mechanical simulations (porous flow and solid mechanics modules, as far as I understood) under dynamic conditions, e.g. seismic actions.
In particular, the u-p formulation is a well-known one in the geotechnical community, which includes the inertial forces due to pore fluid, assumed to move together with the soil skeleton. Is something like that available in MOOSE?
Here's an abstract from Argani & Kowalczyk (2019) showing the governing equations.

Thanks,

N

![Untitled](https://user-images.githubusercontent.com/31536633/219470566-8f82d41a-cd60-4bcd-8988-2dc088ad2a85.png)

**Comment by WilkAndy** [comment, marked as answer]
I would answer that it is possible,but that there is no "user friendly" way of doing this.  By "user friendly", i mean a MOOSE Action where your input file just contains something like

```
[SeismicPhysics]
[]
```

and MOOSE magically builds all the Kernels and Materials for you.  Instead, you'll have to build those Kernels from the dynamic part of TensorMechanics, and the Kernels of PorousFlow.  In fact, you hardly need PorousFlow for this - you could use Framework Kernels - but it's probably advantageous to use PorousFlow if you can.   The reason i suggest using as much "modules" stuff is because of numerical problems.  While your equations are quite simple, they contain hidden numerical difficulties, viz the acceleration terms, which are usually damped numerically (and TensorMechanics offers a few options) and perhaps advection of the fluid, which should usually be stabilised (PorousFlow offers a few options).
